### PR TITLE
Write a blog on upgrading to elm 0.19

### PR DESCRIPTION
**Pull request by saramuse**
Story details: https://app.clubhouse.io/itprotv/story/17873

Hoping for some feedback :-) still need to expand on the "what we like" and would love some suggestions

### CODE REVIEW

File: _posts/Upgrading_Elm.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# Upgrading Elm from 0.18 to 0.19
+by Sara Lichtenstein (@saralich)
+
+[Elm](https://elm-lang.org/) is a front-end, functional language that compiles into JavaScript for seamless web development. When I started at ITProTV, we were using Elm 0.18 for one of our apps. As we worked on that app, we decided it was in our best interest to update our app once Elm 0.19 was released. The creator of Elm released [this upgrade guide](https://github.com/elm/compiler/blob/a968e817e65d30493c344ac96c9c904b19a7f038/upgrade-docs/0.19.md) as well as [this document about what changed](https://elm-lang.org/blog/small-assets-without-the-headache), both were helpful but neither was an end-all answer. In this post, we will go over what didn't go well for us, changes we as a team would suggest, and the wonderful things about Elm 0.19 that make this upgrade worth the potential struggle.
+
+## What Didn't Go Well For Us
```
Comment (tillydray) [inline _posts/Upgrading_Elm.md]: I thought of another one: their debugger no longer works in 0.19. That's not about upgrading to 0.19, it's about being on 0.19, so maybe it's not in scope for this post idk.

File: _posts/Upgrading_Elm.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# Upgrading Elm from 0.18 to 0.19
+by Sara Lichtenstein (@saralich)
+
+[Elm](https://elm-lang.org/) is a front-end, functional language that compiles into JavaScript for seamless web development. When I started at ITProTV, we were using Elm 0.18 for one of our apps. As we worked on that app, we decided it was in our best interest to update our app once Elm 0.19 was released. The creator of Elm released [this upgrade guide](https://github.com/elm/compiler/blob/a968e817e65d30493c344ac96c9c904b19a7f038/upgrade-docs/0.19.md) as well as [this document about what changed](https://elm-lang.org/blog/small-assets-without-the-headache), both were helpful but neither was an end-all answer. In this post, we will go over what didn't go well for us, changes we as a team would suggest, and the wonderful things about Elm 0.19 that make this upgrade worth the potential struggle.
```
Comment (tillydray) [inline _posts/Upgrading_Elm.md]:
```suggestion
[Elm](https://elm-lang.org/) is a front-end, functional language that compiles to JavaScript for seamless web development, and we love it. It's been around for years and in August of 2018 it upgraded to version 0.19. We didn't get around to upgrading our app to 0.19 until January of 2019. This is our experience.

The creator of Elm released [this upgrade guide](https://github.com/elm/compiler/blob/a968e817e65d30493c344ac96c9c904b19a7f038/upgrade-docs/0.19.md) as well as [this document about what changed](https://elm-lang.org/blog/small-assets-without-the-headache), both were helpful but neither was an end-all answer. In this post, we will go over what didn't go well for us, changes we as a team would suggest, and the wonderful things about Elm 0.19 that make this upgrade worth the potential struggle.
```

File: _posts/Upgrading_Elm.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+## What Didn't Go Well For Us
+
+1. Several breaking changes
+   1. Date/Posix
+   1. Navigation (key)
+   1. JSON decoding
+   1. HTTP library
+   1. Various function signatures changed without obvious benefit
+1. With breaking changes comes a broken compiler, which would not be revived until dependencies were sorted through. This is especially noticeable with Elm, whose compiler is very friendly and helpful, so not having it available was both obvious and painful. While that does make some sense, having a working compiler to guide us through how to use new dependencies would have been helpful in figuring out what was broken more efficiently.
+1.  The old documentation became tough to find - the 0.19 documents replaced 0.18 on elm-lang, and after some digging, we were able to find a Github site that contained the 0.18 docs. It was good they were somewhere, but some link between 0.19 and 0.18 would have been helpful.
```
Comment (tillydray) [inline _posts/Upgrading_Elm.md]:
```suggestion
1.  The old documentation became tough to find - the 0.19 documents replaced 0.18 on elm-lang, and after some digging we were able to find a Github site that contained the 0.18 docs, but then that went away and it was some time before we discovered their (hopefully permanent) location [here](https://dmy.github.io/elm-0.18-packages/). It was good they were somewhere, but some link between 0.19 and 0.18 would have been helpful.
```

File: _posts/Upgrading_Elm.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+1. Some elm-community packages were no longer supported, and we chose those over others specifically because they were elm-community and were more likely to be supported, integrated, or at least easy to update.
+1. We had to fork and upgrade some dependencies because it was easier to upgrade them than to choose another dependency, namely [elm-bootstrap](https://github.com/rundis/elm-bootstrap) and [elm-dropdown](https://github.com/sporto/elm-dropdown). That's unfortunate but it happens.
+
+## Potential Upgrade Changes
+
+1. As noted, having a working compiler, especially elm's friendly compiler, makes working through kinks much easier. With 0.19, the compiler and packages have to be upgraded simultaneously. If these were to be separated, allowing for one to be upgraded at a time, the upgrade process would be much smoother in both scenarios.
```
Comment (tillydray) [inline _posts/Upgrading_Elm.md]:
```suggestion
1. As noted, having a working compiler, especially elm's friendly compiler, makes working through kinks much easier. With 0.19, the compiler and packages have to be upgraded simultaneously. If these were separate, allowing for one to be upgraded at a time, the upgrade process would be much smoother in both scenarios.
```

File: _posts/Upgrading_Elm.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+1. While breaking changes can be necessary, it would be nice to have a bridge in-between in order to help acclimate to a breaking change - for example, introducing breaking changes but allowing for compiler compatibility in an 0.18.5 version in order to make the change over from 0.18 to 0.19 smoother.
+1. Make previous version documentation easier to find and make it easier to compare to new version, in case type signatures change. A lot of the documentation available for other languages allows a user to select versions to look at in order to compare between versions.
```
Comment (tillydray) [inline _posts/Upgrading_Elm.md]:
```suggestion
1. Make previous versions' documentation easier to find, and make it easier to compare with the new version. This is super helpful when type signatures change. A lot of the documentation available for other languages allows a user to select versions to look at in order to compare between versions.
```

**Review by tillydray — DISMISSED** [review]
Some mostly grammatical and phrasing changes.

**Review by tfausak — APPROVED** [review]

**Comment by saramuse** [comment]
Did a few fixes to make some of the language more positive and update the author - I don't feel comfortable being the only one listed, as this is mostly not my writing, so I put the team.

**Review by tillydray — APPROVED** [review]

**Review by tillydray — CHANGES_REQUESTED** [review]
We should change the name of the file before posting.

**Review by tillydray — APPROVED** [review]

**Review by Xandrak — APPROVED** [review]
Lookin' good!

**Pull request merged**